shaman/controllers/builds/__init__.py

```diff
@@ -0,0 +1,175 @@
+from projects import ProjectsAPIController # noqa
+from pecan import expose, request, abort
+from shaman.models import Build, Project
+from sqlalchemy import desc
+
+
+class BuildController(object):
+
+    def __init__(self, build_id):
+        self.build_id = build_id
+        self.project = Project.get(request.context['project_id'])
+        if not self.project:
+            # TODO: nice project not found error template
+            abort(404, 'project not found')
+
+    @expose('jinja:build.html')
+    def index(self):
+        build = Build.filter_by(
+            project=self.project,
+            ref=request.context['ref'],
+            sha1=request.context['sha1'],
+            distro=request.context['distro'],
+            distro_version=request.context['distro_version'],
+            arch=request.context['arch'],
+            flavor=request.context['flavor'],
+            build_id=self.build_id
+        ).first()
+        if not build:
+            # TODO: nice project not found error template
+            abort(404, 'build not found')
+
+        return dict(
+            project_name=self.project.name,
+            build=build
+        )
+
+
+class FlavorController(object):
+
+    def __init__(self, flavor_name):
+        self.flavor_name = flavor_name
+        self.project = Project.get(request.context['project_id'])
+        if not self.project:
+            # TODO: nice project not found error template
+            abort(404, 'project not found')
+        request.context['flavor'] = flavor_name
+
+    @expose('jinja:builds.html')
+    def index(self):
+        builds = Build.filter_by(
+            project=self.project,
+            ref=request.context['ref'],
+            sha1=request.context['sha1'],
+            distro=request.context['distro'],
```

Comment:
``distro``, ``distro_version`` and ``arch`` are no longer part of the ``Build`` model